scandium-core/src/test/java/org/eclipse/californium/scandium/dtls/RecordCbcValidationTest.java

```diff
@@ -0,0 +1,937 @@
+ * HmacSHA256-MAC compensation-44-03-all*:  999[252] < 1043 < 1219[209]: 186
+ * HmacSHA256-MAC none compen.-41-06-same:  773[046] <  796 <  973[106]: 187
+ * HmacSHA256-MAC compensation-45-02-all*: 1000[006] < 1040 < 1220[095]: 190
+ * HmacSHA256-MAC compensation-43-04-same: 1008[212] < 1044 < 1229[161]: 192
+ * HmacSHA256-MAC compensation-43-68-all*: 1149[007] < 1180 < 1368[120]: 194
+ * HmacSHA256-MAC compensation-45-66-all*: 1152[105] < 1192 < 1379[250]: 194
+ * HmacSHA256-MAC compensation-40-07-same: 1010[004] < 1048 < 1245[081]: 203
+ * HmacSHA256-MAC compensation-42-69-all*: 1156[110] < 1194 < 1410[138]: 223
+ * HmacSHA256-MAC compensation-42-69-same: 1167[226] < 1207 < 1432[237]: 233
+ * HmacSHA256-MAC none compen.-42-69-same:  769[119] <  796 < 1022[222]: 234
+ * HmacSHA256-MAC compensation-42-05-same: 1016[029] < 1052 < 1279[137]: 235
+ * HmacSHA256-MAC compensation-41-06-same: 1017[113] < 1049 < 1288[192]: 245
+ * HmacSHA256-MAC compensation-41-06-all*: 1008[145] < 1046 < 1284[204]: 247
+ * HmacSHA256-MAC none compen.-40-07-all*:  355[088] <  468 <  938[004]: 547
+ * HmacSHA256-MAC none compen.-42-05-all*:  353[244] <  468 <  953[003]: 559
+ * HmacSHA256-MAC none compen.-41-06-all*:  354[254] <  466 <  953[000]: 568
+ * HmacSHA256-MAC none compen.-44-03-all*:  349[201] <  459 <  952[004]: 569
+ * HmacSHA256-MAC none compen.-45-02-all*:  351[177] <  462 <  971[003]: 586
+ * HmacSHA256-MAC none compen.-43-04-all*:  354[189] <  465 < 1080[004]: 697
+ * HmacSHA256-MAC none compen.-41-70-all*:  355[149] <  603 < 1108[001]: 702
+ * HmacSHA256-MAC none compen.-40-71-all*:  350[206] <  603 < 1108[000]: 706
+ * HmacSHA256-MAC none compen.-43-68-all*:  349[213] <  598 < 1110[004]: 708
+ * HmacSHA256-MAC none compen.-45-66-all*:  352[215] <  604 < 1128[000]: 719
+ * HmacSHA256-MAC none compen.-42-69-all*:  348[180] <  597 < 1123[001]: 725
+ * HmacSHA256-MAC none compen.-44-67-all*:  348[255] <  603 < 1153[000]: 747
+ * (overall 47 min.)
+ * </pre>
+ * 
+ * <table summary="Explanation">
+ * <tr>
+ * <td>MAC compensation</td>
+ * <td>verification with MAC compensation</td>
+ * </tr>
+ * <tr>
+ * <td>MAC none compen.</td>
+ * <td>verification without MAC compensation (old implementation)</td>
+ * </tr>
+ * <tr>
+ * <td>same</td>
+ * <td>always use the same, proper padding length byte</td>
+ * </tr>
+ * <tr>
+ * <td>all*</td>
+ * <td>use all 256 values for the padding length byte</td>
+ * </tr>
+ * </table>
+ * 
+ * Summarize: The old verification implementation without MAC compensation shows
+ * a clear timing differences for "all*". The other benchmarks show smaller
+ * differences, and it seems to be not related to "same" or "all*".
+ * 
+ * Remark: This test contains some more stuff; different padding operators and
+ * different statistic functions. Feel free to create your own benchmarks!
+ */
+@RunWith(Parameterized.class)
+@Ignore
+public class RecordCbcValidationTest {
+
+	private static final int LOOPS = 500;
```

Comment:
You may reduce this number of loops for slower machines.
The benchmark uses 256*LOOPS decrypt calls to measure the time.